ambry-account/src/main/java/com/github/ambry/account/AccountInfoMap.java

```diff
@@ -0,0 +1,267 @@
+        return true;
+      }
+    }
+    return false;
+  }
+
+  /**
+   * Updates the {@code AccountInfoMap} with the input {@link Collection} of {@link Account}s.
+   * @param accounts collection of {@link Account}s to be added.
+   */
+  void updateAccounts(Collection<Account> accounts) {
+    for (Account account : accounts) {
+      Account accountToUpdate = idToAccountMap.get(account.getId());
+      if (accountToUpdate == null) {
+        accountToUpdate = account;
+      } else {
+        accountToUpdate.setStatus(account.getStatus());
+        accountToUpdate.setSnapshotVersion(account.getSnapshotVersion());
+        accountToUpdate.updateContainers(account.getAllContainers());
+      }
+      idToAccountMap.put(account.getId(), accountToUpdate);
+      nameToAccountMap.put(account.getName(), accountToUpdate);
+    }
+  }
+
+  /**
+   * Updates the {@code AccountInfoMap} with the input {@link Collection} of {@link Container}s.
+   * @param containers collection of {@link Container}s to be added.
+   */
+  void updateContainers(Collection<Container> containers) {
+    for (Container container : containers) {
+      addContainer(container.getParentAccountId(), container);
+    }
+  }
+
+  /**
+   * Adds a {@link Container} to the {@link Account}.
+   * @param accountId The id of the parent {@link Account} for this {@link Container}.
+   * @param container {@link Container} to be added.
+   * @throws IllegalArgumentException if {@link Account} with provided id doesn't exist.
+   */
+  void addContainer(short accountId, Container container) {
+    Account parentAccount = idToAccountMap.get(accountId);
+    if (parentAccount == null) {
+      throw new IllegalArgumentException("Account with ID " + accountId + "doesn't exist");
+    }
+    idToAccountMap.get(accountId).updateContainers(Collections.singletonList(container));
+  }
+
+  /**
+   * Gets {@link Container} by its Parent Account id and id.
+   * @param accountId The id of the parent {@link Account} for this {@link Container}.
+   * @param id The id to get the {@link Container}.
+   * @return The {@link Container} with the given id within the parent Account Id, or {@code null} if
+   * such a parent {@link Account} or {@link Container} does not exist.
+   */
+  Container getContainerByIdForAccount(Short accountId, short id) {
+    Account parentAccount = idToAccountMap.get(accountId);
+    return parentAccount == null ? null : parentAccount.getContainerById(id);
+  }
+
+  /**
+   * Gets the last modified time of accounts and containers in this in-memory cache
+   * @return the last modified time of accounts and containers
+   */
+  long getLastModifiedTime() {
+    return lastModifiedTime;
+  }
+
+  /**
+   * Sets the last modified time of accounts and containers in this in-memory cache
+   * @param lastModifiedTime time when the accounts and containers were last updated
+   */
+  void setLastModifiedTime(long lastModifiedTime) {
```

Comment:
Where is this called?

Reply:
This will be called from MySqlAccountService::fetchAndUpdateCache(). Want to set it to max LMT value found in fetched accounts and containers.